contracts/kernel/Kernel.sol

```diff
@@ -193,8 +193,7 @@ contract Kernel is IKernel, KernelStorage, Petrifiable, IsContract, VaultRecover
     */
     function hasPermission(address _who, address _where, bytes32 _what, bytes _how) public view returns (bool) {
         IACL defaultAcl = acl();
-        return address(defaultAcl) != address(0) && // Poor man's initialization check (saves gas)
-            defaultAcl.hasPermission(_who, _where, _what, _how);
+        return defaultAcl.hasPermission(_who, _where, _what, _how); // if acl is not set, it will revert
```

Comment:
⚠️ In this case I don't think we want to revert, but to return `false`, so we still need this case.

Reply:
Great catch :)

Reply:
Actually, almost everywhere else we are reverting in getters when data doesn't exist or it is not initialized. Was there any other reason `kernel#hasPermission` returned false rather than failing?

Reply:
Not sure about this, but the only way `defaultACL` is `0` is if `Kernel` has not been initialized, right? Would it make sense to call `hasPermission` on a non-initialized kernel?
About the getters, that's true at least for out-of-bounds arrays and mappings mimicking arrays. We did that because it's the default compiler behaviour.

Reply:
I think it makes sense for an uninitialized `Kernel` to respond with `false` when asked for permission, since an uninitialized kernel should not allow anything and this is a _read_ operation. An uninitialized `ACL` acts similarly; you can't _write_ any permissions, but all _read_ attempts will return false.

Another reason I'd advocate for it to keep returning only a boolean, rather than throwing, is because `hasPermission()` is part of the `IKernel` interface. We could, of course, add to the interface that one may revert based on certain conditions, but this gets more and more tricky for consumers to handle.

> Actually, almost everywhere else we are reverting in getters when data doesn't exist or it is not initialized.

As @bingen said, this should be only in cases where we mimic array behaviour.